File: frontend/src/contracts/audit.ts
export interface DispatchSummaryItem {
    uuid: string;
    consultation_id: number;
    patient_name: string;
    doctor_name: string;
    issue_date: string; // ISO 8601
    email_sent_at: string | null;
    whatsapp_sent_at: string | null;
}

export interface DispatchAuditResponse {
    items: DispatchSummaryItem[];
    total_count: number;
}

export type DispatchStatus = 'all' | 'pending' | 'sent';

export interface DispatchAuditFilterParams {
    start_date?: string;
    end_date?: string;
    status?: DispatchStatus;
}

File: backend/schemas/audit.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional, List
from enum import Enum

class DispatchSummaryItem(BaseModel):
    uuid: str
    consultation_id: int
    patient_name: str
    doctor_name: str
    issue_date: datetime
    email_sent_at: Optional[datetime]
    whatsapp_sent_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)

class DispatchAuditResponse(BaseModel):
    items: List[DispatchSummaryItem]
    total_count: int

class DispatchStatus(str, Enum):
    ALL = "all"
    PENDING = "pending"
    SENT = "sent"

class DispatchAuditFilterParams(BaseModel):
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    status: Optional[DispatchStatus] = DispatchStatus.ALL
